src/scram.js

```diff
@@ -0,0 +1,167 @@
+import utils from './utils';
+
+async function scramClientProof( authMessage, clientKey, hashName ) {
+    let enc = new TextEncoder();
+
+    const storedKey = await window.crypto.subtle.importKey(
+      "raw",
+      await window.crypto.subtle.digest(hashName, clientKey),
+      { "name": "HMAC", "hash": hashName },
+      false,
+      ["sign"]
+    );
+    const clientSignature = await window.crypto.subtle.sign("HMAC", storedKey, enc.encode(authMessage));
+
+    return utils.xorArrayBuffers(clientKey, clientSignature);
+}
+
+function scramParseChallenge ( challenge ) {
+    /* This function parses the information in a SASL SCRAM challenge response,
```

Comment:
Please put the function docstrings above the function declaration, not inside it.